db/db_test_util.h

```diff
@@ -761,6 +761,17 @@ class OnFileDeletionListener : public EventListener {
   size_t matched_count_;
   std::string expected_file_name_;
 };
+
+class FlushCounterListener : public EventListener {
+ public:
+  std::atomic<int> count{0};
```

Comment:
Very minor: these can be non-public, and naming convention is `count_` and `expected_flush_reason_`.

Reply:
They need to be public, the tests access them directly. I can add setters/getters but it seems overkill.